include/vkd3d_core_interface.idl

```diff
@@ -0,0 +1,37 @@
+/*
+ * * Copyright 2023 Joshua Ashton
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+import "vkd3d_d3d12.idl";
+
+[
+    uuid(18da885c-a7b5-464e-a121-ccb75d4dfc31),
+    object,
+    local,
+    pointer_default(unique)
+]
+interface IVKD3DCoreInterface
+{
+    HRESULT CreateDevice(IUnknown *adapter, D3D_FEATURE_LEVEL minimum_feature_level, REFIID iid, void **device);
```

Comment:
An interface that does not inherit from IUnknown, is that okay? There's no way to AddRef/Release either.

Reply:
Yes, its a global interface we define ourselves and use ourselves so we can do whatever we want. We could make it IUnknown but it would just be a useless allocation without any free path.